applications/visualizer/frontend/src/models/workspace.ts

```diff
@@ -117,4 +132,44 @@ export class Workspace {
         });
         this.updateContext(updated);
     }
+
+    async _initializeActiveDatasets(datasetIds: Set<string>) {
+        const datasets = await fetchDatasets(datasetIds);
+        const updated: Workspace = produce(this, (draft: Workspace) => {
+            draft.activeDatasets = datasets;
+        });
+        const updatedWithNeurons = await this._getAvailableNeurons(updated);
+        this.updateContext(updatedWithNeurons);
+    }
+
+    async _getAvailableNeurons(updatedWorkspace: Workspace): Promise<Workspace> {
+        try {
+            const neuronPromises = Object.keys(updatedWorkspace.activeDatasets).map(datasetId =>
+                DatasetsService.getDatasetNeurons({dataset: datasetId})
+            );
+
+            const neuronArrays = await Promise.all(neuronPromises);
+
+            return produce(updatedWorkspace, (draft: Workspace) => {
+                // Reset the neuronsAvailable map
+                draft.availableNeurons = {};
+
+                // Populate neuronsAvailable with neurons from all active datasets
+                neuronArrays.forEach(neurons => {
```

Comment:
I wonder if you can flat and put in a set before the `forEach`, otherwise, you might treat the same neurons many times.